### ISSUE TITLE
cross validation

### ISSUE THREAD

**Issue by lironghua318** [KevinMusgrave/powerful_benchmarker#3]
num_training_partitions: 10
num_training_sets: 5
Translation:

The test set consists of classes with labels in [num_labels * start_idx, num_labels * (start_idx+size)]. Note that if we set start_idx to 0.9, the range would wrap around to the beginning (0.9 to 1, 0 to 0.4).
The remaining classes will be split into 10 equal sized partitions.
5 of those partitions will be used for training. In other words, 5-fold cross validation will be performed, but the size of the partitions will be the same as if 10-fold cross validation was being performed.

**if i use cross validation what parameters will be tune  and where can i set?**

**Comment by KevinMusgrave** [comment]
Great question! With the latest few commits, you can now tune hyperparameters using Bayesian optimization:

1. Append ```~BAYESIAN~``` to any parameters you want to tune, followed by a lower and upper bound in square brackets. 

2. Specify the ```--bayesian_optimization_init_points``` and ```--bayesian_optimization_n_iter``` flags.

3. Instead of using ```run.py```, use ```run_bayesian_optimization.py```.

Note that each step in bayesian optimization will create a new experiment folder, which contains all the usual config files, models etc.

Here is an example of how I use the bayesian optimization script:
```
python run_bayesian_optimization.py --bayesian_optimization_init_points 8 --bayesian_optimization_n_iter 80 \
--loss_funcs~OVERRIDE~ {metric_loss: {MultiSimilarityLoss: {alpha~BAYESIAN~: [0.01, 100], beta~BAYESIAN~: [0.01, 100], base~BAYESIAN~: [0, 1]}}} \
--mining_funcs~OVERRIDE~ {post_gradient_miner: {MultiSimilarityMiner: {epsilon~BAYESIAN~: [0, 1]}}} \
--experiment_name cub200_test5050_multi_similarity_with_ms_miner \
--root_experiment_folder /home/tkm45/experiments/cub200_test5050_multi_similarity_with_ms_miner/
```

[(This info comes from the new section in the readme)](https://github.com/KevinMusgrave/powerful_benchmarker#bayesian-optimization-to-tune-hyperparameters)

Let me know if that works!

**Comment by KevinMusgrave** [comment]
Now you can also append ```~LOG_BAYESIAN~``` if you want the optimizer to explore x linearly, in the expression 10^x.

**Comment by lironghua318** [comment]
i will try~ thx

**Comment by KevinMusgrave** [comment]
Due to a bug in scikit-learn (which the BayesianOptimization package depends on), I have updated the bayesian optimization script to use the Ax package instead (https://github.com/facebook/Ax). The only usability change is that when you use ~LOG_BAYESIAN~, the range should be specified like it is for ~BAYESIAN~ (i.e. not as an exponent).
